transport/middleware.go

```diff
@@ -1,19 +1,79 @@
 package transport
 
 import (
-	"fmt"
+	"errors"
 	"net/http"
+	"strings"
+
+	reperr "release-manager/repository/errors"
+	svcmodel "release-manager/service/model"
+	neterr "release-manager/transport/errors"
+	"release-manager/transport/utils"
+
+	httpx "go.strv.io/net/http"
 )
 
-func (h *Handler) recoverPanic(next http.Handler) http.Handler {
+func (h *Handler) auth(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		defer func() {
-			if err := recover(); err != nil {
-				w.Header().Set("Connection", "close")
-				WriteServerErrorResponse(w, r, fmt.Errorf("%s", err))
+
+		authorizationHeader := r.Header.Get(httpx.Header.Authorization)
+
+		if authorizationHeader == "" {
+			r = utils.ContextSetUser(r, svcmodel.AnonUser)
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		headerParts := strings.Split(authorizationHeader, " ")
+		if len(headerParts) != 2 || headerParts[0] != "Bearer" {
+			WriteInvalidAuthenticationResponse(w, neterr.ErrInvalidBearer)
+			return
+		}
+
+		tokenString := headerParts[1]
+
+		user, err := h.UserSvc.GetForToken(r.Context(), tokenString)
+		if err != nil {
+			switch {
+			case errors.Is(err, reperr.ErrUserAuthenticationFailed):
+				WriteInvalidAuthenticationResponse(w, err)
+				return
+			default:
+				WriteServerErrorResponse(w, err)
+				return
 			}
-		}()
+		}
+
+		r = utils.ContextSetUser(r, &user)
```

Comment:
The service User object is now stored in the context.
I believe that storing the User struct in the context is acceptable, the object is not too big.